### ISSUE TITLE
[NcRichText] Level down headings in markdown

### ISSUE THREAD

**Issue by ShGKme** [nextcloud-libraries/nextcloud-vue#7743]
Having high-level headings (H1..H3) in `NcRichText` is not correct semantically in a general case as it doesn't follow the page structure. For example, there is already `H1` on the page and there must be only a single one. `H2` are also present on the page usually and `H3` are used for smaller sections.

Thus using `NcRichText` with Markdown for user content is an accessibility issue.

The proposal is to level down the headings starting the user content from `<h4>`. 

The following text

```md
# Heading 1
## Heading 2
### Heading 3
#### Heading 4
##### Heading 5
###### Heading 6
```

must be rendered as

```html
<h4>Heading 1</h4>
<h5>Heading 2</h5>
<h6>Heading 3</h6>
<h6>Heading 4</h6>
<h6>Heading 5</h6>
<h6>Heading 6</h6>
```

This makes using `h4-h6` headings not very useful for the end user. But users often use low-level heading only to have a heading with a "smaller" font size even when from the structure it should have been high-level heading. Thus, we can not only use the lower level headings tags, but also the lower level heading styles, which look good in a file comment or Talk message.

**Comment by ShGKme** [comment]
One possible solution: adjust the `createElement` method, check the `type` for being a `<hN>` tag and level down the `N`.

Alternative: add a new plugin that changes levels down the actual level of the heading after parsing MD.
